contracts/governance/compatibility/IGovernorCompatibilityBravo.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "../IGovernor.sol";
+
+/**
+ * @dev Interface extension that adds missing functions to the {Governor} core to provide `GovernorBravo` compatibility.
+ *
+ * _Available since v4.3._
+ */
+interface IGovernorCompatibilityBravo is IGovernor {
+    /**
+     * @dev Proposal structure from Compound Governor Bravo
+     */
+    struct Proposal {
```

Comment:
This causes an issue. If you put the mapping in the structure, then you get an error when compiling the accessor

```function proposals(uint256 proposalId) public view virtual override returns (Proposal memory)```

it tells you that a proposal that contains a mapping cannot live in memory.

Reply:
I understand we may have reasons to remove it, but does it have any observable consequence of the kind I suggested?

Reply:
When looking at the ABI, there might be an issue with the return type indeed, but not the one we expected.

This is the ABI entry from a governor alpha/bravo:
```
{
  "constant":true,
  "inputs":[
    {"internalType":"uint256","name":"","type":"uint256"}
  ],
  "name":"proposals",
  "outputs":[
    {"internalType":"uint256","name":"id","type":"uint256"},
    {"internalType":"address","name":"proposer","type":"address"},
    {"internalType":"uint256","name":"eta","type":"uint256"},
    {"internalType":"uint256","name":"startBlock","type":"uint256"},
    {"internalType":"uint256","name":"endBlock","type":"uint256"},
    {"internalType":"uint256","name":"forVotes","type":"uint256"},
    {"internalType":"uint256","name":"againstVotes","type":"uint256"},
    {"internalType":"bool","name":"canceled","type":"bool"},
    {"internalType":"bool","name":"executed","type":"bool"}
  ],
  "payable":false,
  "stateMutability":"view",
  "type":"function"
},
```

and this is the one for our compatibility layer

```
{
  "inputs":[
    {"internalType":"uint256","name":"proposalId","type":"uint256"}
  ],
  "name":"proposals",
  "outputs":[
    {
      "components":[
        {"internalType":"uint256","name":"id","type":"uint256"},
        {"internalType":"address","name":"proposer","type":"address"},
        {"internalType":"uint256","name":"eta","type":"uint256"},
        {"internalType":"address[]","name":"targets","type":"address[]"},
        {"internalType":"uint256[]","name":"values","type":"uint256[]"},
        {"internalType":"string[]","name":"signatures","type":"string[]"},
        {"internalType":"bytes[]","name":"calldatas","type":"bytes[]"},
        {"internalType":"uint256","name":"startBlock","type":"uint256"},
        {"internalType":"uint256","name":"endBlock","type":"uint256"},
        {"internalType":"uint256","name":"forVotes","type":"uint256"},
        {"internalType":"uint256","name":"againstVotes","type":"uint256"},
        {"internalType":"uint256","name":"abstainVotes","type":"uint256"},
        {"internalType":"bool","name":"canceled","type":"bool"},
        {"internalType":"bool","name":"executed","type":"bool"}
      ],
      "internalType":"struct IGovernorCompound.Proposal",
      "name":"",
      "type":"tuple"
    }
  ],
  "stateMutability":"view",
  "type":"function"
}```

It appears that the default accessor added by solidity with `mapping (uint => Proposal) public proposals;` doesn't just remove the mapping, it also removes the array accesses. That is why they also have the `getActions` function that would otherwise look like a duplicate:

```
function getActions(uint proposalId) public view returns (address[] memory targets, uint[] memory values, string[] memory signatures, bytes[] memory calldatas)
```

I'm going to change our implementation of `proposals` to fix this newly found incompatibility